IdentityCore/src/webview/systemWebview/session/MSIDSystemWebViewControllerFactory.m

```diff
@@ -100,10 +100,18 @@ + (MSIDWebviewType)availableWebViewTypeWithPreferredType:(MSIDWebviewType)prefer
         return authSession;
     }
     
+#if defined TARGET_OS_VISION && TARGET_OS_VISION
+    return nil;
+#endif
+
+#if !MSID_EXCLUDE_SYSTEMWV
     return [[MSIDSafariViewController alloc] initWithURL:startURL
                                         parentController:parentController
                                         presentationType:presentationType
                                                  context:context];
+#else
+    return nil;
+#endif
```

Comment:
you could check #if !MSID_EXCLUDE_SYSTEMWV && !(defined TARGET_OS_VISION && TARGET_OS_VISION) to return SafariViewController and return nil otherwise

Reply:
Makes sense, good suggestion. Updated